### PR TITLE
Unmap external dtb

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Fixes https://github.com/OP-TEE/optee_os/issues/4278

### CODE REVIEW

File: core/arch/arm/kernel/boot.c [hunk 1]
```diff
@@ -75,7 +75,9 @@ DECLARE_KEEP_PAGER(sem_cpu_sync);
 #ifdef CFG_DT
 struct dt_descriptor {
 	void *blob;
+#ifdef CFG_EXTERNAL_DTB_OVERLAY
```
Comment (etienne-lms) [inline core/arch/arm/kernel/boot.c:78]: Change seems unrelated. Fixes a build issue?

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: core/arch/arm/kernel/boot.c (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/boot.c:78]: No, it doesn't fix a build issue. I can move this into a separate patch if that's preferred.

Comment (jforissier) [inline core/arch/arm/kernel/boot.c:78]: Separate commit?

**Review by jforissier — COMMENTED** [review]
- For commit "core: add core_mmu_rem_mapping()": with or without my comment addressed,
`Acked-by: Jerome Forissier <jerome@forissier.org>`
- For commit "core: remove temporary external DT mapping", without the `#ifdef` chunk:
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

File: core/arch/arm/include/mm/core_mmu.h [hunk 2]
```diff
@@ -614,6 +614,8 @@ static inline bool core_mmu_is_shm_cached(void)
 		(TEE_MATTR_CACHE_CACHED << TEE_MATTR_CACHE_SHIFT);
 }
 
+TEE_Result core_mmu_rem_mapping(enum teecore_memtypes type, void *addr,
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/include/mm/core_mmu.h]: OK

File: core/arch/arm/kernel/boot.c (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/boot.c:78]: Done

**Comment by jenswi-linaro** [comment]
Tags applied to all commits except the new "core: ifdef frag_id member in struct dt_descriptor".

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` for commit "core: ifdef frag_id member in struct dt_descriptor".

**Comment by jenswi-linaro** [comment]
Tag applied

**Comment by jenswi-linaro** [comment]
Ready to merge?

**Comment by jforissier** [comment]
> Ready to merge?

Yep, missed that one :-/

**Comment by jenswi-linaro** [comment]
Thanks.
